p2p/src/authenticated/lookup/actors/listener.rs

```diff
@@ -34,6 +39,7 @@ pub struct Actor<
     address: SocketAddr,
     stream_cfg: StreamConfig<C>,
     rate_limiter: RateLimiter<NotKeyed, InMemoryState, E, NoOpMiddleware<E::Instant>>,
+    registered_ips: Option<Arc<RwLock<HashSet<IpAddr>>>>,
```

Comment:
The rationale for making this a RwLock<HashSet> is that it is expected to be more efficient than sending a message to the tracker on each iteration.

This is particularly true because we will rarely update this list.

It is probably even better to just pass a copy here whenever updated to avoid all locking.

Reply:
That being said, checking a channel each loop isn't great either